libs/ledger-live-common/src/families/bitcoin/wallet-btc/explorer/index.ts

```diff
@@ -163,55 +160,41 @@ class BitcoinLikeExplorer implements IExplorer {
     return parseFloat(fees["relay_fee"]);
   }
 
-  async getPendings(address: Address, nbMax?: number): Promise<TX[]> {
-    const params: ExplorerParams =
-      this.explorerVersion === "v2"
-        ? {
-            noToken: "true",
-          }
-        : {
-            no_token: "true",
-          };
-    if (!this.disableBatchSize) {
-      params.batch_size = nbMax || 1000;
-    }
-    const res = await this.fetchTxs(address, params);
-    const pendingsTxs = res.txs.filter((tx) => !tx.block);
+  async getPendings(address: Address, nbMax = 1000) {
+    const params: ExplorerParams = {
+      no_token: "true",
+      batch_size: !this.disableBatchSize ? nbMax : undefined,
+      block_hash: undefined,
+    };
+    const pendingsTxs = await this.fetchPendingTxs(address, params);
     pendingsTxs.forEach((tx) => this.hydrateTx(address, tx));
     return pendingsTxs;
   }
 
-  async fetchTxs(
+  async fetchTxs(address: Address, params: ExplorerParams): Promise<TX[]> {
+    const url = `/address/${address.address}/txs`;
+    // TODO add a test for failure (at the sync level)
+    const client = await this.client.acquire();
+    const response = await client.client.get(url, {
+      params,
+    });
+    await this.client.release(client);
+    const pendingTxs = await this.fetchPendingTxs(address, params);
+    Array.prototype.push.apply(response.data.data, pendingTxs);
```

Comment:
two problems here:
- response.data.data shouldn't be altered "in place". make a copy array
- as this can fail (apply limits), do a for loop or just use `reponse.data.data.concat(pendingTxs)` 🤔

Reply:
I know this looks dirty but response.data.data array is never used again

If I am not mistaken apply limits are only for the parts that are passed as arguments (pendingTxs here), I don't think we can ever reach 65k pending transactions on a customer account. This is exactly what I had before and concat creates another array which is why scanAccount was timing out on the bot

But if you find something that is suitable performance wise other than Array.prototype.push.apply or push with the spread operator (those 2 are similar but have the 65k parameters limit)

Reply:
But I changed a lot of things before running the bot again and noticing an upgrade in performance so I might try a last time the concat option before merging to know if it was the culprit

Reply:
https://github.com/airbnb/javascript/issues/243
https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki